docs/utilities/idempotency.md

```diff
@@ -206,6 +206,11 @@ In this example, we have a Lambda handler that creates a payment for a user subs
 
 Imagine the function executes successfully, but the client never receives the response due to a connection issue. It is safe to retry in this instance, as the idempotent decorator will return a previously saved response.
 
+!!! warning "Making JSON APIs Idempotent"
+    To ensure JSON structured APIs are fully idempotent we must ensure the idempotent key is a valid JSON object.
+
+    To do this we can [customize the default behaviour](#customizing-the-default-behavior) with *event_key_jmespath* and the [JMESPath built-in function](/utilities/jmespath_functions) *powertools_json()*.
```

Comment:
```suggestion
    To alter this behaviour, we can use the [JMESPath built-in function](/utilities/jmespath_functions) *powertools_json()* to treat the payload as a JSON object rather than a string.
```